packages/react-dom-bindings/src/server/ReactDOMLegacyServerStreamConfig.js

```diff
@@ -16,6 +16,12 @@ export opaque type PrecomputedChunk = string;
 export opaque type Chunk = string;
 export opaque type BinaryChunk = string;
 
+export function scheduleMicrotask(callback: () => void) {
+  throw new Error(
+    'Legacy React Server builds should not be using scheduleMicrotask. This is a bug in React.',
+  );
```

Comment:
Should environments that previously scheduled all work sync just continue to do sync work? I Should probably rename to `scheduleWorkInCurrentTask` since it otherwise lies about what it does.

I made this error because we have no legacy builds of flight which is currently the only place this new API is used but this feels pretty inconsistent to me.